### PR TITLE
Implementation of getShippingUnitTotal

### LINKED ISSUE #11515: [Bug] Shipment's getShippingUnitTotal always return 0
**Sylius version affected**: master / any

**Description**  
During implementing shipping rate calculator (which have different rates based on order total), realized `getShippingUnitTotal` at `Shipment` always return `0`.

https://github.com/Sylius/Sylius/blame/master/src/Sylius/Component/Shipping/Model/Shipment.php#L219

But I expect to have shipment items (order items) total there.

This bug's age is about 7 years, as `getShippingUnitTotal` never used by Sylius or covered by tests, when similar `getShippingUnitCount` is valid as used by `PerUnitRateCalculator`. I mean its another `+1` for https://github.com/Sylius/Sylius/issues/10813 :)

**Possible Solution**  
Calculate total from units.

I will provide PR if you guys OK with solution.

### PR DESCRIPTION
**Pull request by igormukhingmailcom**
| Q               | A
| --------------- | -----
| Branch?         | 1.7
| Bug fix?        | yes
| New feature?    | yes
| BC breaks?      | not sure
| Deprecations?   | not sure
| Related tickets | fixes #11515
| License         | MIT

### CODE REVIEW

File: src/Sylius/Component/Shipping/Model/Shipment.php [hunk 1]
```diff
@@ -165,7 +165,7 @@ public function getShippingUnitCount(): int
 
     public function getShippingUnitTotal(): int
     {
-        return 0;
+        throw new \Exception('You have to override and implement getShippingUnitTotal method');
```
Comment (igormukhingmailcom) [inline src/Sylius/Component/Shipping/Model/Shipment.php:168]: I'm not sure about this particular line. Maybe we should return `0` as it was before, but before that add deprecation error?

Comment (vvasiloi) [reply src/Sylius/Component/Shipping/Model/Shipment.php:168]: I would revert it.
Those who use the shipping component standalone, either care about it and already overrode the method, or don't care about it and won't be affected anyway.
And for Sylius Standard you already provided an implementation in the core model.

**Review by vvasiloi — CHANGES_REQUESTED** [review]
Hey @igormukhingmailcom, I remember seeing it more than a year ago thinking: finally, someone implemented this. Back then, I though that it doesn't make sense to leave it return 0, but now I think it's better to leave it that way.
Would you like to give a second chance?
If so, don't forget to rebase.

**Comment by NoResponseMate** [comment]
Hi!

I've opened a [replacement PR](https://github.com/Sylius/Sylius/pull/15025); just rebased and applied Victor's suggestion, therefor I'm closing this.

Cheers :beers:

**Comment by vvasiloi** [comment]
Thanks @NoResponseMate!

**Pull request closed without merging**